Support simple ${project_path} syntax in options

jjgod opened the pull request:
Simple ${project_path} can be convenient for per-project options like:

```
"sublimeclang_options":
[
  "-I${project_path}/../../../../build/texk",
  "-I${project_path}/../../../../build/texk/web2c",
  "-I${project_path}/../..",
  "-I${project_path}/.."
]
```

It's simpler to specify than more versatile ${project_path:filename} syntax.

jjgod committed:
Support simple ${project_path} syntax in options

quarnster commented:
The api `window.project_file_name()` is ST3 only so I cannot accept this pull request as is.

jjgod commented:
Sorry, I didn't realize that, any alternative for ST2?

jjgod committed:
fixup! Support simple ${project_path} syntax in options

Since Sublime Text 2 provides no project file path information from
window, we have to figure it out by checking each fodler in window.folders()
and return the first folder that contains a .sublime-project file as our
best guess.

jjgod commented:
Added a fallback support for ST2.

quarnster commented:
Thanks!
